onnxruntime/core/common/cpuid_info.cc

```diff
@@ -24,7 +24,7 @@
 #if _WIN32
 #define HAS_WINDOWS_DESKTOP WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 #endif
-#if (defined(CPUIDINFO_ARCH_X86) || defined(CPUIDINFO_ARCH_ARM)) && defined(CPUINFO_SUPPORTED) && (!_WIN32 || HAS_WINDOWS_DESKTOP)
+#if (defined(CPUIDINFO_ARCH_X86) || defined(CPUIDINFO_ARCH_ARM)) && defined(CPUINFO_SUPPORTED) && (!_WIN32 || defined(HAS_WINDOWS_DESKTOP))
```

Comment:
I'm confused by the error "unmatched parenthesis: missing", as all the parens *appear* balanced. I worry about replacing an `ifdef` with an `if` and vice versa in general (or an `if` with a `if defined(...)` in this case) because the two are not identical, depending on what is set distantly elsewhere in the build. It *should* be legal to use a definition in an `if` whether it's defined or not because undefined macros just yield 0 (https://en.cppreference.com/w/cpp/preprocessor/conditional - "The result is 1 if the identifier was defined as a macro name, otherwise the result is ​0​").

- Does the PVS error happen when `_WIN32` is true or false, or both? 
- Does it also happen when wrapping with just an extra set of parens (but no `defined`)? e.g. `(!_WIN32 || (HAS_WINDOWS_DESKTOP))`

I sign off, as it seems fine in this case - just wondering details. TY.

Reply:
@tiagoshibata - look good to you?

Reply:
It also took me a while to debug the error for the very confusing error message.

- It happened on Windows, not on Linux.
- Good question, let me come back to you about it. Long story short, I can only run PVS every so often and it takes a few hours, but I'll come back to you as soon as I can check that :)

Reply:
Oh, thought it was an easy error to check (e.g. displayed in the IDE or something as you're editing like VS does). Never mind if it takes a few hours - the CI should already be done by then :b.